### PR TITLE
feat(chartcuterie): refactor image buidler

### PR DESCRIPTION
**Pull request by sentaur-athena**
Final goal: support chartcuterie in emails ([specs](https://www.notion.so/sentry/Chartcuterie-Improvements-Technical-Design-Doc-3dd20eac5c9c494ab0c6bebd0d0a6dad?pvs=4#1ae0c446e5f94feb82cd88b34c055053))

This PR: Refactoring image building code out of slack into `IssueAlertImageBuilder` so we can share with email.

PRs:
1. Add caching
2. Add image to email

### CODE REVIEW

File: src/sentry/integrations/issue_alert_image.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                return image_url
+        except Exception as e:
+            self.logger.exception(
+                "issue_alert_chartcuterie_image.failed",
+                extra={"exception": e, "group_id": self.group.id},
+            )
+            sentry_sdk.capture_exception()
+        return None
+
+    def _get_endpoint_regression_image_url(self) -> str | None:
```
Comment (sentaur-athena) [inline src/sentry/integrations/issue_alert_image.py:57]: Nothing is new below this line. All copy paste from `image_block_builder`

File: src/sentry/integrations/issue_alert_image_builder.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                return image_url
+        except Exception as e:
+            self.logger.exception(
+                "issue_alert_chartcuterie_image.failed",
+                extra={"exception": e, "group_id": self.group.id},
+            )
+            sentry_sdk.capture_exception()
+        return None
+
+    def _get_endpoint_regression_image_url(self) -> str | None:
```
Comment (sentaur-athena) [inline src/sentry/integrations/issue_alert_image_builder.py:56]: Nothing is new below this line. All copy paste from image_block_builder

File: src/sentry/integrations/issue_alert_image_builder.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+from sentry.models.group import Group
+from sentry.snuba.referrer import Referrer
+from sentry.types.integrations import ExternalProviderEnum
+from sentry.utils import metrics
+from sentry.utils.performance_issues.detectors.utils import escape_transaction
+
+
+class IssueAlertImageBuilder:
+    def __init__(
+        self, group: Group, provider: ExternalProviderEnum, logger: logging.Logger
```
Comment (ykamo001) [inline src/sentry/integrations/issue_alert_image_builder.py:26]: Can we have a default logger so calling side doesn't have to make one?

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/integrations/issue_alert_image_builder.py (hunk 2 above)
Comment (ykamo001) [reply src/sentry/integrations/issue_alert_image_builder.py:56]: Super helpful comment for reviewing, thank you for this haha

Comment (ykamo001) [reply src/sentry/integrations/issue_alert_image_builder.py:56]: Wish GH just had a "please mark this as moved" or something to help reviewers

File: src/sentry/integrations/issue_alert_image_builder.py (hunk 3 above)
Comment (sentaur-athena) [reply src/sentry/integrations/issue_alert_image_builder.py:26]: We can. I was thinking email might want their own logger vs. slack. But the error would be the same so what you're suggesting makes more sense 😅

Comment (ykamo001) [reply src/sentry/integrations/issue_alert_image_builder.py]: Oh no, sorry, I meant just add a default for the argument, but you can keep the parameter!

So like
```
def __init__(
        self, group: Group, provider: ExternalProviderEnum, logger: logging.Logger = _DEFAULT_LOGGER
```

Comment (ykamo001) [reply src/sentry/integrations/issue_alert_image_builder.py]: What you had makes sense, allowing the caller to pass in their own logger if they choose

File: src/sentry/integrations/issue_alert_image_builder.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+    def get_image_url(self) -> str | None:
+        try:
+            metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
+            if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
+                image_url = self._get_endpoint_regression_image_url()
+            elif self.group.issue_type == ProfileFunctionRegressionType:
+                image_url = self._get_function_regression_image_url()
+
```
Comment (ykamo001) [inline src/sentry/integrations/issue_alert_image_builder.py:43]: ahh we forgot the else/base case

File: src/sentry/integrations/issue_alert_image_builder.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        self.provider = provider
+        self.cache_key = f"chartcuterie-image:{self.group.id}"
+        self.tags = {
+            "provider": self.provider,
+            "issue_category": self.group.issue_category,
+        }
+
+    def get_image_url(self) -> str | None:
+        try:
+            metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
```
Comment (ykamo001) [inline src/sentry/integrations/issue_alert_image_builder.py:38]: probably just need a `image_url = None` here

**Pull request merged**